### ISSUE TITLE
ruff check with file read from stdin does not use options from pyproject.toml when --stdin-filename argument is used

### ISSUE THREAD

**Issue by ipvych** [astral-sh/ruff#17405]
### Summary

# Steps to reproduce:
```shell
mkdir repro
cd repro
echo '[project]
requires-python = ">=3.11"
[tool.ruff.lint]
extend-select = ["UP"]' > pyproject.toml

mkdir src
cd src
echo -e 'from typing import Callable\nCallable' > repro.py

ruff check --stdin-filename stdin - < repro.py
```

# Observed behavior:
Ruff reported no issues found in file

# Expected behavior:
Pyupgrade plugin that was enabled in pyproject.toml should have reported `Import from collections.abc instead: Callable` error.  Error is correctly reported when I remove `--stdin-filename <anything>` flag from ruff command line or when I run `ruff check repro.py` directly.

### Version

ruff 0.11.5 (7186d5e9a 2025-04-10)

**Comment by ipvych** [comment]
I expected `--stdin-filename` to only affect the text that is displayed as file name (the `-` part of `-:1:1: UP035 [*] Import from `collections.abc` instead: `Callable``) when file is passed via stdin and nothing else. 

I want to change that part to `stdin` so output is similar to that of flake8 allowing me to reuse existing tooling that expects flake8 output by just replacing command to ruff. Specifically in Emacs there is a builtin variable specifying command to use for linting code that I want to make just work with ruff without messing up with regexps for parsing the output of the command.
